syntax = "proto3";

package MoviesRecommender;

service MoviesRecommender {
    rpc GetRecommendations (GetRecommendationsRequest) returns (GetRecommendationsResponse);
}

message ViewContext {
    string movie_id = 1;
}

message GetRecommendationsRequest {
    string user_id = 1;
    repeated ViewContext views = 2;
}

message MovieContext {
    string movie_id = 1;
}

message GetRecommendationsResponse {
    repeated MovieContext movies = 1;
}
